### PR TITLE
gc_worker: add failpoint & unit test for #16372

### PR DESCRIPTION
**Pull request by zyguan**
Signed-off-by: zyguan <zhongyangguan@gmail.com>

### What problem does this PR solve?

Problem Summary: #16372 fixed a bug of gc worker but didn't add test for that. 

### What is changed and how it works?

What's Changed: This PR add unit test as well as a failpoint for #16372 .

How it Works: Instead of using the failpoint, the unit test push errors to `done` channel dirctly, because it take at least `gcWorkerTickInterval` (1 min) for gc worker calling `runGCJob`, which is unacceptable for CI.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- None

### Release note 
- No release note.

### CODE REVIEW

**Comment by jackysp** [comment]
PTAL @MyonKeminta

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -490,6 +490,9 @@ func (w *GCWorker) calculateNewSafePoint(now time.Time) (*time.Time, error) {
 }
 
 func (w *GCWorker) runGCJob(ctx context.Context, safePoint uint64, concurrency int) error {
+	failpoint.Inject("mockRunGCJobFail", func() {
```
Comment (cfzjywxk) [inline store/tikv/gcworker/gc_worker.go:493]: Is this used for `automated-tests`, we could link the test in this change

Comment (zyguan) [reply store/tikv/gcworker/gc_worker.go:493]: It's not used by automated-tests currently. I added it for release testing.

**Pull request merged**